timeinterval/timeinterval.go

```diff
@@ -25,34 +25,60 @@ import (
 	"time"
 
 	"gopkg.in/yaml.v2"
+
+	"github.com/prometheus/alertmanager/types"
 )
 
 // Intervener determines whether a given time and active route time interval should mute outgoing notifications.
 // It implements the TimeMuter interface.
 type Intervener struct {
 	intervals map[string][]TimeInterval
+	marker    types.Marker
+}
+
+func (i *Intervener) Active(groupKey string, alerts []*types.Alert, names []string, now time.Time) (bool, error) {
+	matches, err := i.inIntervals(names, now)
+	if err != nil {
+		return false, err
+	}
+	for _, a := range alerts {
```

Comment:
The marking here is incorrect and will need to be fixed if we agree to continue having both `Active` and `Mute`.